### ISSUE TITLE
[JSON] JSON.stringify(undefined) panics

### ISSUE THREAD

**Issue by croraf** [boa-dev/boa#509]
In Chrome it returns "undefined".

But in the spec https://tc39.es/ecma262/#sec-json.stringify it says:
![image](https://user-images.githubusercontent.com/19698636/85065050-cf61b600-b1ac-11ea-8418-b526eb61750c.png)

In boa it panics:
```
thread 'main' panicked at 'internal error: entered unreachable code: Symbols and Undefined JSON Values depend on parent type', boa/src/builtins/value/mod.rs:264:17
```

**Comment by n14little** [comment]
I'll look into this.

**Comment by croraf** [comment]
Behavior in chrome (we should probably have tests for the below cases):
![image](https://user-images.githubusercontent.com/19698636/85086830-11eeb700-b1dc-11ea-8152-271fd8552a76.png)

**Comment by n14little** [comment]
@croraf For your first case, there is [this test](https://github.com/boa-dev/boa/blob/a7cffe3c7c6fc3f0588a6fc1c84bd7dfeddd4bfd/boa/src/builtins/json/tests.rs#L28-L40) on master.
For your second case, there is [this test](https://github.com/boa-dev/boa/blob/a7cffe3c7c6fc3f0588a6fc1c84bd7dfeddd4bfd/boa/src/builtins/json/tests.rs#L186-L199) on master.
The undefined case is being addressed in #512 
And for your last case, there is [this test](https://github.com/boa-dev/boa/blob/a7cffe3c7c6fc3f0588a6fc1c84bd7dfeddd4bfd/boa/src/builtins/json/tests.rs#L186-L199) on master.

**Comment by croraf** [comment]
Oh. I now see NOTE 5 in the spec exactly explains how `JSON.stringify(undefined)` should be returned.
